selectors.go

```diff
@@ -32,153 +29,215 @@ const (
 	}
-	return copyMap
-}
 
-func ChainIdFromSelector(chainSelectorId uint64) (uint64, error) {
-	for k, v := range evmChainIdToChainSelector {
-		if v.ChainSelector == chainSelectorId {
-			return k, nil
-		}
+	chainDetail, ok = testSelectorToChainDetailsMap[chainSelectorId]
+	if ok {
+		return chainDetail.ChainID, nil
 	}
-	return 0, fmt.Errorf("chain not found for chain selector %d", chainSelectorId)
+
+	return "0", fmt.Errorf("chain not found for chain selector %d", chainSelectorId)
 }
 
-func SelectorFromChainId(chainId uint64) (uint64, error) {
-	if chainSelectorId, exist := evmChainIdToChainSelector[chainId]; exist {
-		return chainSelectorId.ChainSelector, nil
+func SelectorFromChainIdAndFamily(chainId string, family string) (uint64, error) {
+	// if family is missing use EVM as default
+	if family == "" {
+		family = FamilyEVM
+	}
+
+	selectorMap, exist := chainIDToSelectorMapForFamily[family]
+	if !exist {
+		return 0, fmt.Errorf("chain selector map not found for family %v", family)
+	}
+
+	selector, exist := selectorMap[chainId]
+	if !exist {
+		return 0, fmt.Errorf("chain selector not found for chainID %v, family %v", chainId, family)
 	}
-	return 0, fmt.Errorf("chain selector not found for chain %d", chainId)
+
+	return selector, nil
 }
 
-func NameFromChainId(chainId uint64) (string, error) {
-	details, exist := evmChainIdToChainSelector[chainId]
+func NameFromChainIdAndFamily(chainId string, family string) (string, error) {
+	// if family is missing use EVM as default
+	if family == "" {
+		family = FamilyEVM
+	}
+
+	selectorMap, exist := chainIDToSelectorMapForFamily[family]
+	if !exist {
+		return "", fmt.Errorf("chain family not found for chain %v, family %v", chainId, family)
+	}
+
+	selector, exist := selectorMap[chainId]
+	if !exist {
+		return "", fmt.Errorf("chain selector not found for chain %v, family %v", chainId, family)
+	}
+
+	details, exist := selectorToChainDetails[selector]
 	if !exist {
-		return "", fmt.Errorf("chain name not found for chain %d", chainId)
+		return "", fmt.Errorf("chain details not found for chain %v, family %v", chainId, family)
 	}
-	if details.ChainName == "" {
-		return strconv.FormatUint(chainId, 10), nil
+
+	// when name is missing use chainID
+	if details.Name == "" {
+		return chainId, nil
 	}
-	return details.ChainName, nil
+	return details.Name, nil
 }
 
-func ChainIdFromName(name string) (uint64, error) {
-	for k, v := range evmChainIdToChainSelector {
-		if v.ChainName == name {
-			return k, nil
-		}
+func ChainIdFromNameAndFamily(name string, family string) (string, error) {
+	// if family is missing use EVM as default
+	if family == "" {
+		family = FamilyEVM
 	}
-	chainId, err := strconv.ParseUint(name, 10, 64)
-	if err == nil {
-		if details, exist := evmChainIdToChainSelector[chainId]; exist && details.ChainName == "" {
-			return chainId, nil
+
+	for _, v := range selectorToChainDetails {
+		if v.Name == name && family == v.Family {
+			return v.ChainID, nil
```

Comment:
Has our schema enforced anywhere that name, family paid must be unique?
I know practically we use unique names, but nothing currently prevents from someone naming say 2 different EVM chains as same?

Just wondering, if we leave that logic as it-is, or add some validation?

Reply:
I guess we could add a unit-test to ensure that name+family combination should not have duplicates, if others agree with this.